### PR TITLE
Add support for Forgejo/Gitea remote repos

### PR DESCRIPTION
**Pull request by nephros**
This adds support for remote SCM hosts running gitea or forgejo software, similar to the gitlab and github integration we already have (except this is not using GraphQL).

Tested only with codeberg.org (forgejo), and by running curl against https://try.gitea.io.  

API docs can be accessed by browsing to `<instance host>/api/swagger` for both gitea and forgejo, e.g:

 - https://codeberg.org/api/swagger
 - https://try.gitea.io/api/swagger

A new API key will have to be generated for at least codeberg.org so Github can use it in the build Action.
The format of the `TOKEN_FORGEJO` for github actions follows that of the Gitlab implementation:

    "gitea.example.com:XXXX|forgejo.example.com:YYYY|...|siteN:tokenN"

Quoting being essential here! (The github workflow breaks otherwise because of pipes in a shell script!)

s.a. https://github.com/sailfishos-chum/sailfishos-chum-gui/pull/44

### CODE REVIEW

**Comment by Olf0** [comment]
Thank you very much for this addition.

I have to admit that it took me a bit of research to fully understand, why you treated `forgejo` and `gitea` as one: [`Forgejo`](https://forgejo.org/) ([source code](https://codeberg.org/forgejo/forgejo)) is a fork of [`Gitea`](https://about.gitea.com/products/gitea) ([source code](https://github.com/go-gitea/gitea), [add-ons](https://gitea.com/gitea)) in December 2022, which in  turn is a fork of [`Gogs`](https://gogs.io/) ([source code](https://github.com/gogs/gogs)) in November 2016.

All three are written in Go, and `forgejo` and `gitea` obviously still share the same token (i.e. "API key") format and APIs.

@nephros, is there a specific reason why you did not mention Gogs?  E.g. does it not still share the same token format and APIs?  Or was it just due to not being aware of the common, original basis these three `git`-based web-frontends share?  As you can see Gogs' SCM, it is still under development (albeit slowly).

We also should document that `forgejo` and `gitea` currently means the same with regard to token format and APIs as used by the SailfishOS:Chum GUI app.  But that sure does not have to be addressed by this PR.  Currently I even lack an idea where to document this best; I will try to think about it and look at extant documentation to find an appropriate place during the next weekend.

> Tested only with codeberg.org (forgejo), and by running curl against https://try.gitea.io/.

I think that is fine, because [`codeberg.org`](https://codeberg.org/) is more or less `forgejo`'s public reference server.

For Gogs you may try running `curl` against [`try.gogs.io`](https://try.gogs.io/) (uses a valid x.509 certificate for a different domain!), or try one of the methods (local installation or running a container) from the [section "Easy to install"](https://gogs.io/) (the ["Install Now" button](https://gogs.io/) resolves to [`docs/installation`](https://gogs.io/docs/installation), which primarily describes the requirements for a local installation).

P.S.: WRT code review I kindly ask @piggz and/or @rinigus, if one or both of you can spare the time; but actually one should first review and react to the latest comments for the much [older PR \#266](https://github.com/sailfishos-chum/sailfishos-chum-gui/pull/266), because it has been stalled since December 2023.

**Comment by Olf0** [comment]
> I found API docs here:
> https://github.com/gogs/docs-api

Cool, I somehow missed these.

> Could be it just works.

Basically the whole Gogs story if only for these cases: "Just works" or "works with minimal adaptions".  Any other case would put it outside of having a common denominator with Forgejo and Gitea, any way.

> The thing with all the self-hosted things is …

IMO that is fine; how else would one address that easily?

> Scratch that, try.gogs.io is not working ATM.

As stated (I later expanded that statement to clarify it):

> > For Gogs you may try running `curl` against [`try.gogs.io`](https://try.gogs.io/) (uses a valid x.509 certificate for a *different* domain: It might require to use `curl -k` (= `--insecure`) or download the certificate chain in `PEM` format in a browser and use `curl --cacert <foo>.pem`), or try one of the methods (local installation or running a container) from the [section "Easy to install"](https://gogs.io/).

**Comment by Olf0** [comment]
I was just about to walk through [your nice step-by-step guide](https://github.com/sailfishos-chum/sailfishos-chum-gui/pull/271#issuecomment-2075031196) how to create a token for ForgeJo / Gitea using `codeberg.org`, but GitHub does not allow to look at extant tokens (just to overwrite them with new content), so I went through the code again to determine the format used by the SailfishOS:Chum-GUI app … only to see later that [you](https://github.com/sailfishos-chum/sailfishos-chum-gui/pull/271#issue-2261350361) and [rinigus](https://github.com/sailfishos-chum/sailfishos-chum-gui/pull/44) documented this nicely: I would have missed the quoting.

Side note: Still this made me realise that the whole scheme is not well maintainable, when multiple GitLab or ForgeJo / Gitea instances are accessed.  Either one saves the "secret" (well not so much here) tokens offline (hardly feasible with multiple maintainers), or one retrieves them by `strings`, or one crates all tokens for the GitLab or ForgeJo / Gitea instances anew, because GitHub does not allow to extend an extant token string (only allows for overwriting it).  The latter is also not really feasible, because one does not now which and how many instances are already in a token string, when intending to add a new one; this leaves a single, awkward method as viable.

#### TL;DR
A secret `TOKEN_FORGEJO` with the content `"codeberg.org:<token>"` is now set, created according to [your excellent guidance](https://github.com/sailfishos-chum/sailfishos-chum-gui/pull/271#issue-2261350361).

**Comment by Olf0** [comment]
Fine, I think we are all set; as my call for a proper code review (which I cannot provide) [by the other co-maintainers](https://github.com/sailfishos-chum/sailfishos-chum-gui/pull/271#issuecomment-2076019321) was unsuccessful, do you @nephros want to look over it again, before I merge this PR?
